x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/literal/interval/Intervals.java

```diff
@@ -97,6 +97,16 @@ public static TemporalAmount of(Source source, long duration, TimeUnit unit) {
         }
     }
 
+    public static Interval<?> from(String isoValue, DataType dataType) {
+        if (isYearMonthInterval(dataType)) {
```

Comment:
I would move the `from` methods from the classes here as well. Mainly because those classes have behavior and a contract (same methods, generics) while this utility class only static methods.
Also doing a type check to then do `TypeIChecked.doSomething` defeats the whole purpose of putting said method in the source class as it's not generic.
Further more the methods are one liners..